src/main/sensors/esc_sensor.c

```diff
@@ -183,95 +183,95 @@ uint8_t escTelemetryFrameStatus(void)
+        if (debugMode == DEBUG_ESC_SENSOR) debug[0] = escSensorMotor+1;
 
-        motorDmaOutput_t * const motor = getMotorDmaOutput(escTelemetryMotor);
+        motorDmaOutput_t * const motor = getMotorDmaOutput(escSensorMotor);
         motor->requestTelemetry = true;
-        escTelemetryTriggerState = ESC_TLM_TRIGGER_PENDING;
+        escSensorTriggerState = ESC_SENSOR_TRIGGER_PENDING;
     }
 
     if (escTriggerTimestamp + ESC_REQUEST_TIMEOUT < currentTimeMs) {
         // ESC did not repond in time, retry
         timeoutRetryCount++;
         escTriggerTimestamp = currentTimeMs;
-        escTelemetryTriggerState = ESC_TLM_TRIGGER_READY;
+        escSensorTriggerState = ESC_SENSOR_TRIGGER_READY;
 
         if (timeoutRetryCount == 4) {
             // Not responding after 3 times, skip motor
-            escTelemetryData[escTelemetryMotor].skipped = true;
+            escSensorData[escSensorMotor].skipped = true;
             selectNextMotor();
         }
 
-        if (debugMode == DEBUG_ESC_TELEMETRY) debug[1]++;
+        if (debugMode == DEBUG_ESC_SENSOR) debug[1]++;
     }
 
     // Get received frame status
-    uint8_t state = escTelemetryFrameStatus();
+    uint8_t state = escSensorFrameStatus();
 
-    if (state == ESC_TLM_FRAME_COMPLETE) {
+    if (state == ESC_SENSOR_FRAME_COMPLETE) {
         // Wait until all ESCs are processed
-        if (escTelemetryMotor == getMotorCount()-1) {
+        if (escSensorMotor == getMotorCount()-1) {
             escCurrent = 0;
             escConsumption = 0;
             escVbat = 0;
 
             for (int i = 0; i < getMotorCount(); i++) {
-                if (!escTelemetryData[i].skipped) {
-                    escVbat =  i > 0 ? ((escVbat + escTelemetryData[i].voltage) / 2) : escTelemetryData[i].voltage;
-                    escCurrent = escCurrent + escTelemetryData[i].current;
-                    escConsumption = escConsumption + escTelemetryData[i].consumption;
+                if (!escSensorData[i].skipped) {
+                    escVbat =  i > 0 ? ((escVbat + escSensorData[i].voltage) / 2) : escSensorData[i].voltage;
+                    escCurrent = escCurrent + escSensorData[i].current;
+                    escConsumption = escConsumption + escSensorData[i].consumption;
                 }
             }
         }
 
-        if (debugMode == DEBUG_ESC_TELEMETRY) debug[2] = escVbat;
-        if (debugMode == DEBUG_ESC_TELEMETRY) debug[3] = escCurrent;
+        if (debugMode == DEBUG_ESC_SENSOR) debug[2] = escVbat;
+        if (debugMode == DEBUG_ESC_SENSOR) debug[3] = escCurrent;
 
         selectNextMotor();
-        escTelemetryTriggerState = ESC_TLM_TRIGGER_READY;
+        escSensorTriggerState = ESC_SENSOR_TRIGGER_READY;
     }
 
     if (escTriggerLastTimestamp + 10000 < currentTimeMs) {
         // ESCs did not respond for 10 seconds
         // Disable ESC telemetry and fallback to onboard vbat sensor
-        freeEscTelemetryPort();
+        freeEscSensorPort();
```

Comment:
I'm with @hydra here. If your ESC sensor stops working mid flight then there is probably something wrong and you should probably land as soon as possible to check it out. So you want to know about it. So I think getting a low voltage indicator when your ESC sensor fails is the correct behaviour.

Also, since this is new code, we want user feedback on how often the ESC sensor fails, so we don't want to hide it from the user. I expect failure is infrequent, in which case the whole issue is moot. But we won't know failure is infrequent unless, at least in 3.1, we make sure the user is aware of failure.